src/main/java/upb/dice/twitter/TweetSearchApplication.java

```diff
@@ -1,156 +1,77 @@
-            List<Status> queryTweets = queryResult.getTweets();
-            long maxId;
-            if (counter == 0) {
-                maxId = queryResult.getMaxId();
-                //Location based
-                if (query.getQuery() == null) {
-                    try (BufferedWriter out = new BufferedWriter(new FileWriter("Latitude_Longitude_MaxID.txt", true))) {
-                        out.append(String.valueOf(lat)).append(":").append(String.valueOf(lon)).append(":").append(String.valueOf(maxId));
-                        out.append('\n');
+        CommandLine finalCmd = cmd;
+        assert finalCmd != null;
+            timer.schedule(new TimerTask() {
+                @Override
+                public void run() {
+                    TweetExplorer searchTweets = new TweetExplorer();
+                    if (finalCmd.hasOption("h")) {
+                        HelpFormatter formatter = new HelpFormatter();
+                        formatter.printHelp("Command Line Helper", options);
+                        System.exit(0);
                     }
-                }
-                //Keyword Based
-                else {
-                    String key = query.getQuery();
-                    try (BufferedWriter out = new BufferedWriter(new FileWriter("Keyword_MaxID.txt", true))) {
-                        out.append(key).append(":").append(String.valueOf(maxId));
-                        out.append('\n');
+                    //if there is a value for keybased, get it
+                    if (finalCmd.hasOption("k")) {
+                        String[] searchArgs = finalCmd.getOptionValues("k");
+                        String key = searchArgs[0];
+                        long periodSeconds = Long.parseLong(searchArgs[1]);
+                        checkPeriod[0] = periodSeconds * 60 * 60 * 1000;
+                        try {
+                            searchTweets.keywordQuery(key);
+                        } catch (IOException | TwitterException | InterruptedException e) {
+                            e.printStackTrace();
+                        }
                     }
+                    if (finalCmd.hasOption("l")) {
+                        String[] searchArgs = finalCmd.getOptionValues("l");
+                        double latitude = Double.parseDouble(searchArgs[0]);
+                        double longitude = Double.parseDouble(searchArgs[1]);
+                        double radius = Double.parseDouble(searchArgs[2]);
+                        long periodSeconds = Long.parseLong(searchArgs[3]);
+                        checkPeriod[0] = periodSeconds * 60 * 60 * 1000;
+                        try {
+                            searchTweets.locationQuery(latitude, longitude, radius);
+                        } catch (IOException | TwitterException | InterruptedException e) {
+                            e.printStackTrace();
+                        }
+                    }
```

Comment:
I would suggest to keep the parsing of parameters outside of the timer. Simply because they have to be parsed only once and the handling of errors is much easier to handle before the timer is started (e.g., imagine the case that the user forgot to add the radius).